servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TerminalSignalConsumers.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+import static java.util.Objects.requireNonNull;
+
+final class TerminalSignalConsumers implements TerminalSignalConsumer {
```

Comment:
Does this have to `implement TerminalSignalConsumer`?

Also, this does not follow the typical pattern where in such classes we provide static factory methods instead of exposing the concrete implementations.

Reply:
> Does this have to implement `TerminalSignalConsumer`?

No, this is a leftover from intermediate iteration. This upper-level class used to be `RunnableTerminalSignalConsumer`. Thanks for the catch.

Reply:
Btw, because `TerminalSignalConsumer` has all its methods as `default` noop impl, it's very easy to forget about them. Was bitten by this twice. Should we remove default implementations? If users don't need all 3 signals they can use alternative operators. If they need only 2 signals, it's not a big deal to provide a noop impl for the signal you are not interested in,